pkg/controller/jobframework/setup.go

```diff
@@ -67,24 +69,17 @@ func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
 			if err != nil {
 				return fmt.Errorf("%s: %w: %w", fwkNamePrefix, errFailedMappingResource, err)
 			}
-			if _, err = mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
+			if _, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
 				if !meta.IsNoMatchError(err) {
 					return fmt.Errorf("%s: %w", fwkNamePrefix, err)
 				}
 				logger.Info("No matching API in the server for job framework, skipped setup of controller and webhook")
+				go waitForAPI(context.Background(), mgr, logger, gvk, func() {
+					log.Info(fmt.Sprintf("API now available, starting controller and webhook for %v", gvk))
+					setupControllerAndWebhook(mgr, gvk, fwkNamePrefix, cb, opts...)
```

Comment:
One question here is what happens if a controller/webhook errors on startup? Should kueue kueue restart or should it log the error and continue on?

Reply:
Originally, the error would be logged and we would continue on. - I wouldn't like to restart Kueue based on controller/webhook startup errors as in the event of consistent/repetitive errors on one framework would cause Kueue to never start. With that said, the waitForAPI() takes care of ensuring the API exists before starting the controller/webhook.